scripts/extensions/auto-tagging-widget/src/ImageTaggingComponent/ImageTaggingComponent.tsx

```diff
@@ -0,0 +1,278 @@
+import * as React from "react";
+import { ISuperdesk } from "superdesk-api";
+import { ToggleBoxNext } from "superdesk-ui-framework";
+
+interface ITag {
+  title: string;
+  type: string;
+  pubStatus: boolean;
+  weight: number;
+}
+
+interface IImage {
+  imageUrl: string;
+  thumbnailUrl: string;
+  id?: string;
+  headline?: string;
+  caption?: string;
+  credit?: string;
+  byline?: string;
+  source?: string;
+  dateCreated?: string;
+  archivedTime?: string;
+}
+
+interface IProps {
+  superdesk: ISuperdesk;
+  data: any;
+  style?: React.CSSProperties;
+}
+
+const gridContainerStyle: React.CSSProperties = {
+  display: "grid",
+  gridTemplateColumns: "3fr 1fr",
+  gridColumnGap: "4px",
+  maxHeight: "320px",
+  overflow: "hidden",
+};
+
+const gridItemLeftStyle: React.CSSProperties = {
+  width: "100%",
+  height: "100%",
+  maxHeight: "inherit",
+};
+
+const gridItemRightStyle: React.CSSProperties = {
+  width: "100%",
+  height: "100%",
+  maxHeight: "inherit",
+};
+
+const imageContentStyle: React.CSSProperties = {
+  width: "100%",
+  maxHeight: "inherit",
+  overflowY: "auto",
+  display: "grid",
+  gridRowGap: "4px",
+};
+
+const imageStyle: React.CSSProperties = {
+  width: "100%",
+  height: "auto",
+};
+
+const imageWrapperStyle: React.CSSProperties = {
+  cursor: "pointer",
+  maxHeight: "100%",
+};
+
+const selectedCardStyle: React.CSSProperties = {
+  display: "flex",
+  flexDirection: "column",
+  height: "100%",
+  overflow: "hidden",
+  boxShadow: "1px 3px 6px 0 rgba(0,0,0,0.2)",
+  width: "100%",
+  borderRadius: "4px",
+};
+
+const cardStyle: React.CSSProperties = {
+  boxShadow: "1px 3px 6px 0 rgba(0,0,0,0.2)",
+  width: "100%",
+  borderRadius: "2px",
+};
+
+let isMounted: boolean = false;
+
+const ImageTaggingComponent = (props: IProps) => {
+  const { superdesk, data, style } = props;
+  const { httpRequestJsonLocal } = superdesk;
+
+  const [showImages] = React.useState<boolean>(true);
```

Comment:
Use class based components. We're not using hooks in the codebase.

Reply:
This is solved. Modified the functional component to the class component. Thanks for the feedback!